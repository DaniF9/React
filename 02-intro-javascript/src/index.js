const persona =  {
    nombre: 'Daniel',
    apellido: 'Fragoso',
    edad: 23,
};


console.log(persona) // Esto se puede imprimir normal
console.log({persona}) // Si tenemos mas objeto podemos ver a cual estamos llamando
console.table(persona) // Esto nos entrega una salida en consola pero con formato de tabla

