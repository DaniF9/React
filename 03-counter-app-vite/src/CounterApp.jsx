import PropTypes from 'prop-types';
// Doumentacion oficial de react
export const CounterApp  =({value}) => {

    const handleAdd  = (event,newValue) => {
        console.log(event);
    }

    return(
        <>
        <h1>CounterApp</h1>
        <h2>{ value }</h2>
        <button onClick={  handleAdd }>
            +1
        </button>
        </>
    )
}

CounterApp.prototype = {
    value:PropTypes.number.isRequired
}