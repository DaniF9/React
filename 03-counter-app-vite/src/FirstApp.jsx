import PropTypes from 'prop-types';
/*const newMessage = {
    message: 'Hola mundo',
    tittle:'Daniel'
}
*/

/*
const getResult = (a,b) =>{
    return a*b;
}
*/
export const FirstApp = ({
  tittle,
  subTittle = 'Daniel fragoso',
  name
}) => {

  /*
  if (!tittle) {
    throw new Error('El tittle no existe');
    
  }
  */
  return (
    <>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      {/* <h1>{getResult(1,4)}</h1> */}
      <h1 data-testid="test-title">{tittle}</h1>
      <p>{subTittle}</p>
      <p>{name}</p>
    </>
  )
}

//export default FirstApp


//Las protypes se definen de la siguiente manera

FirstApp.propTypes = {
  tittle:PropTypes.string.isRequired,
  subTittle:PropTypes.string.isRequired
}

FirstApp.defaultProps = {  // Estas son propiedades por defecto 
  //tittle: "No hay titulo",
  subTittle: "No hay subtitulo",
  name:"Daniel Fragoso"
}

