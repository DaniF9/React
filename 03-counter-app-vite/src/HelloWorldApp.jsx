/*
function App() {

    return <h1>Hola mundo</h1>
    
}

export default App;
*/



export const App = () => {
  return (
    <div>
      Hello Wordl
    </div>
  )
}

export default App
