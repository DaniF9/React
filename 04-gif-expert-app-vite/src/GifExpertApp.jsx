import { useState } from "react" //Esto es para utilizar el usestate
import {AddCategory,GifGrid} from "./components";




export const GifExpertApp = () => {

  // cuando queremos almacenar info y esa cambia el html necesitamos usar un hook
  const [categories, setCategories]=useState(['One Punch']); 
  //console.log(categories);

  // const addCategory = () => {
  //   if (!categories.includes('Valorant')) {
  //     setCategories([...categories,'Valorant']);
  //   }
  //   else{
  //     alert("Ya haz agregado esa categoria");
  //   }
    
  // }

  const addCategory = (newCategory) => {

    if (categories.includes(newCategory)) return; //Aqui validamos si existe la categoria
    //setCategories(['Valorant', ...categories]);
    setCategories([newCategory, ...categories]); //El setcategories es de nuestro hook
  }
  return (
    <>
    {/* Titulo */}
      <h1>GiftExpertApp</h1>

      {/* Input */}

     <AddCategory
     // Aqui lo que haremos en emitir un un evento padre ya que lo que se busca es mandar el valor
      onNewCategory = {value => addCategory(value)} //Aqui mandamos a llamar a la funcion 
     //setCategories={setCategories}
     /> 

      {/* Listado de gif */}
      {/* <button onClick={addCategory}>Agregar Categoria</button> */}
      {/* <ol>
        {categories.map(category => {
          return <li key={category}>{category}</li>
          })
          
        }
      </ol> */}

      
        {
          categories.map((category) =>(  // Aqui podemos poner nuestor nuevo componente creado llamado GifGrid
            // <div key = {category}>
            //   <h3>{category}</h3>
            //   <li>{category}</li>
            // </div> ))

              <GifGrid 
              key = {category} 
              category={category}
              />

            ))
            
        
        }
      





          {/* Gif item */}
    </>
  )
}
