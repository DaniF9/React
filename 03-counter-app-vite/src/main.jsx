import React from 'react';    // Estos dos import se ocupan para renderizar la aplicacion
import ReactDOM from 'react-dom/client'; // Esto es por que estamos en web


// Estas ya son para exportar
//import {App} from './HelloWorldApp';
import {FirstApp} from './FirstApp'
import './styles.css';

/*
function App() {  // Esto se usa para hacer los componentes  "componentes basados en funciones"

    
    return <h1>Hola Mundo !!!</h1> // Esto era como lo que se tenia de la sig manera document.createElement...
}
*/

ReactDOM.createRoot(document.getElementById('root')).render(   // Renderizamos el componente en donde lo estemos ejecutando 
    <React.StrictMode>
        <FirstApp/>  
    </React.StrictMode>
);