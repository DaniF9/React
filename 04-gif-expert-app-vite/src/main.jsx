import React from 'react';
import ReactDOM from 'react-dom/client';
import {GifExpertApp} from './GifExpertApp';
import './style.css';


//https://es.legacy.reactjs.org/docs/faq-structure.html
//https://es.legacy.reactjs.org/docs/faq-structure.html

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp/>
  </React.StrictMode>,
)
