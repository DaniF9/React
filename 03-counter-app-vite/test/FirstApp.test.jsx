import { render } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en FirsApp', () => {
    test('Debe de hacer match con el snapshot', () => {

        const tittle = 'Hola soy Daniel';
        const{container} = render( <FirstApp tittle={tittle}/>);    //Render lo que hace es renderizar el componente en memoria
        
        expect(container).toMatchSnapshot() // Esto nos creo la carpeta snapshot
      
    });
    
    test('Debe de mostrar el titulo en un h1', () => {
      
        const tittle = 'Hola soy Daniel';
        const{container, getByText} = render( <FirstApp tittle={tittle}/>)

        expect(getByText(tittle)).toBeTruthy();


        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(tittle);
    })
    
  
})
