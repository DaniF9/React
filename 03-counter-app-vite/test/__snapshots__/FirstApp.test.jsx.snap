// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirsApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy Daniel
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Daniel Fragoso
  </p>
</div>
`;
